### ISSUE TITLE
Virus; Trojan.Gen.MBT

### ISSUE THREAD

**Issue by MaxFun** [niXman/mingw-builds-binaries#56]
Norton 360 found a the virus Trojan.Gen.MBT in your download mingwInstaller.exe
I downloaded from https://github.com/niXman/mingw-builds-binaries 
See Norton's report below

Filename: padlock.dll
Threat name: Trojan.Gen.MBTFull Path: C:\Users\wmcre\mingw64\opt\lib\engines-3\padlock.dll

____________________________

____________________________

On computers as of 
10/12/2023 at 2:31:20 AM

Last Used 
10/12/2023 at 2:31:20 AM

Startup Item 
No
Launched 
No
Threat type: Virus. Programs that infect other programs, files, or areas of a computer by inserting themselves or attaching themselves to that medium.

____________________________

padlock.dllThreat name: Trojan.Gen.MBT
Locate

Few Users
Fewer than 50 users in the Norton Community have used this file.

New
This file was released 10 days  ago.

High
This file risk is high.

____________________________

Source: External Media

____________________________

File Actions

File: C:\Users\wmcre\mingw64\opt\lib\engines-3\padlock.dllBlocked

____________________________

File Thumbprint - SHA:
Not available
File Thumbprint - MD5:
Not available

**Comment by Ragudos** [comment]
Wow. Thank you for this. It's a good thing I checked the issues. I ran the file in VirusTotal:

![image](https://github.com/niXman/mingw-builds-binaries/assets/133567781/8e40b49b-30c1-4225-8d17-7d5e54f01b00)

How come the link in the README file did not show it?

**Comment by MaxFun** [comment]
Not sure what file the link is pointing to or if the file is packed such that the scanner does not detect it. When I downloaded mingwInstaller.exe, Norton said it was OK, see below. But after I ran it,  Norton detected the virus. 

Filename: mingwInstaller.exe
Full Path: C:\Users\wmcre\Downloads\mingwInstaller.exe

____________________________

____________________________


Developers 
Not Available

Version 
Not Available

Identified 
10/13/2023 at 6:18:21 AM

Last Used 
Not Available

Startup Item 
No

____________________________


Few Users
Hundreds of users in the Norton Community have used this file.

Mature
This file was released 10 months  ago.

Good
Norton has given this file a good rating.


____________________________


Source File: 
mingwInstaller.exe

____________________________

Performance

____________________________

Avg. Resource Usage: Moderate
Avg. CPU Usage: Low
Avg. Memory Usage: Moderate

____________________________


File Thumbprint - SHA:
68214ff3d9ddd74538d7d96001173c952284b4c6b62608f6c3fcc447feca1a5d
File Thumbprint - MD5:
c777d8b09b8fe4995f8f53ca59e97604

**Comment by niXman** [comment]
guys, mingwInstaller is a separate project hosted on: https://github.com/Vuniverse0/mingwInstaller
and as I can see the executable file has not changed since `Dec 11, 2022`: https://github.com/Vuniverse0/mingwInstaller/releases/tag/1.2.0

**Comment by niXman** [comment]
Moreover, VirusTotal says that there are no problems: https://www.virustotal.com/gui/file/68214ff3d9ddd74538d7d96001173c952284b4c6b62608f6c3fcc447feca1a5d

**Comment by MaxFun** [comment]
Norton said the same thing till I ran it. I guess the bad guy are really good at hiding their tracks

**Comment by nenin-sc** [comment]
Eset Endpoint (corporative) - no reaction
Eset Antivirus (home) - no reaction

**Comment by MaxFun** [comment]
Was Eset Endpoint & Eset Antivirus only used to check "mingwInstaller.exe" before it was run?
If so, Norton only found "Trojan.Gen.MBTFull" after I ran "mingwInstaller.exe" and not before

**Comment by nenin-sc** [comment]
I tried to run it under Endpoint.  
"The “Gen” in the name indicates that it is a generic detection, meaning that the specific characteristics of the threat may not be fully identified or classified."  Source: https://howtofix.guide/trojan-gen-mbt/

**Comment by matenestor** [comment]
No, it definitely does not look like a false positive! Even Windows Defender is telling me it is a trojan malware. I put it to VirusTotal as well and 23 vendors marked it as a malware (compared to 10 in the screenshot above).
@niXman The problem is that you are testing `mingwInstaller.exe`, but the malicious code is in the file `padlock.dll`.

I downloaded [x86_64-13.2.0-release-win32-seh-msvcrt-rt_v11-rev0.7z](https://github.com/niXman/mingw-builds-binaries/releases/download/13.2.0-rt_v11-rev0/x86_64-13.2.0-release-win32-seh-msvcrt-rt_v11-rev0.7z).
The file is at `x86_64-13.2.0-release-win32-seh-msvcrt-rt_v11-rev0\mingw64\opt\lib\engines-3\padlock.dll`

Here is my result from VirusTotal.

[https://www.virustotal.com/gui/file/82cff646163444ac981fbd9279f503f35f3f9068c23f57d46ea30d17f40c8bf5/details](https://www.virustotal.com/gui/file/82cff646163444ac981fbd9279f503f35f3f9068c23f57d46ea30d17f40c8bf5/details)

![image](https://github.com/niXman/mingw-builds-binaries/assets/44382264/9353d53c-fea7-40f0-ac6d-3712247b9a99)

**Comment by nenin-sc** [comment]
Hmmm...
Endpoint reacted on x86_64-13.2.0-release-win32-seh-msvcrt-rt_v11-rev0\mingw64\opt\lib\engines-3\padlock.dll  as suspicious.
And x86_64-13.2.0-release-win32-seh-ucrt-rt_v11-rev0\mingw64\opt\lib\engines-3\padlock.dll  is OK for him.
No comments.

**Comment by niXman** [comment]
this is definitely a false positive.
this issue can be closed.

**Comment by niXman** [comment]
and? how can I affect this?
I'm not an antivirus software developer...

**Comment by niXman** [comment]
guys, does anyone know where this dll comes from?
what project is it part of?

**Comment by nenin-sc** [comment]
> guys, does anyone know where this dll comes from? what project is it part of?
I suppose this is just beginning of the end of the "free access" to the msvcrt.

**Comment by starg2** [comment]
> guys, does anyone know where this dll comes from? what project is it part of?

openssl

**Comment by ANauzet** [comment]
Hello,
My antivirus found a virus in the gcov.exe from the x86_64 MSVCRT and UCRT version (and also in the padlock.dll)
Impossible to use this at work :-(

**Comment by nenin-sc** [comment]
> Hello, My antivirus found a virus in the gcov.exe from the x86_64 MSVCRT and UCRT version (and also in the padlock.dll) Impossible to use this at work :-(

Just checked with Virustotal: no problems with gcov, padlock marked by AI of "Bkav Pro".
